src/backend/utils/misc/guc-file.l

```diff
@@ -423,6 +443,9 @@ ProcessConfigFile(GucContext context)
 							(errmsg("parameter \"%s\" changed to \"%s\"",
 									item->name, item->value)));
 			}
+
+			if (return_changed)
+				changed_gucs = lappend(changed_gucs, pstrdup(item->name));
```

Comment:
How about returning a list of GUCs, rather than their names?
```suggestion
				{
					struct config_generic *gconf = find_option(item->name, false, elevel);
					if (gconf != NULL)
						changed_gucs = lappend(changed_gucs, gconf);
				}
```